### DISCUSSION TITLE
What are my options for intercepting GET requests

### DISCUSSION THREAD

**Discussion by k0l0ssus** [directus/directus#8364] [Archive]
Hullo: After looking at #3736 and related discussions, I understand that Directus doesn't support any incarnation of hooking into `item.read.before`. I'm interested in any approach that lets me safely intercept GET requests for some collections, and possibly transform or abort them based on some criteria. What would you recommend? Thanks.

**Comment by joselcvarela** [comment]
Hello @k0l0ssus 
Since you are referring to GET requests, I assume you are using the  REST API. So the first thing that comes to my mind is express middlewares.
You can setup those by using custom hooks. As you can see here: https://github.com/directus/directus/blob/51148c11d6496b16a8be29860c6afa3ee4b942ba/api/src/app.ts#L91
We are emitting an `middlewares.init.after` event, so you can listen to it and use `app` to register your middleware.

Since our middlewares are already applied at that stage, you already have access to parsed `req.accountability` which can told some info about the logged user, if you need.

**Reply by k0l0ssus** [reply]
Thanks for answering @joselcvarela ! If I'm reading this correctly, the `middlewares.init.after` will be fired before any `items` event handling. In there, I have access to the full REST request and I can do whatever I want with it? Is there any documentation/examples on the `middlewares.init` event you could point me to please? The official docs is pretty silent on this feature. Thanks again!

**Reply by benhaynes** [reply]
CC: @alesvaupotic for docs

**Reply by joselcvarela** [reply]
@k0l0ssus 
Sorry for the typo above, I meant to reference this code:
 https://github.com/directus/directus/blob/51148c11d6496b16a8be29860c6afa3ee4b942ba/api/src/app.ts#L162
 
 Yes, you are right.
 Yeah, maybe we could add some examples, but here is one (only for showing purposes, does not work):
 ```js
module.exports = function registerHook() {
  return {
    "middlewares.init.after": async function ({ app }) {
      app.use(function (req, res, next) {
        if (!req.url.startsWith("/assets")) return next();

       if(req.body.test) return next()
       
       await doSomething(req.body);
       
       if (req.body.error) return next(new Error());

        return next();
      });
    },
  };
};
```

**Reply by k0l0ssus** [reply]
Noice; thanks!
